doc/changelog.md

```diff
@@ -13,12 +13,17 @@
 
 <h3>Breaking changes</h3>
 
+* ``AllocOp``, ``DeallocOp`` have now (only) value semantics. In the frontend, the last 
+  quantum register is deallocated instead of the first one.
+  [(#360)](https://github.com/PennyLaneAI/catalyst/pull/360)
```

Comment:
This is interesting, because it doesn't represent a breaking change for users. Only if you were to develop at the compiler level would this be a breaking change, but then a lot more of our changes would be as well 🤔 

For now, I might be tempted to list this under improvements (more in line with existing changelog), but it is tricky. Once we know about people developing against the compiler directly, we could revisit this strategy.

It is also important to list why this change was made / what it enables.

Reply:
I will add it to improvement and add some more explanation.